### PR TITLE
Minor refactorings

### PR DESCRIPTION
**Pull request by cpuguy83**
Mostly just re-arranging things, fixing some doc strings.
The biggest thing is making that reconciliation loop easier to read.

### CODE REVIEW

File: vkubelet/helpers.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+// populateSecretsAndConfigMapsInEnv populates Secrets and ConfigMap into environment variables
+func (s *Server) populateSecretsAndConfigMapsInEnv(pod *corev1.Pod) error {
```
Comment (robbiezhang) [inline vkubelet/helpers.go]: shall we change the function name to populateEnvironmentVariables as it supposes to handle the downward API too, in the future.

File: vkubelet/pod.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	for _, pod := range pods {
+		if pod.Status.Phase == corev1.PodSucceeded ||
+			pod.Status.Phase == corev1.PodFailed ||
+			pod.Status.Reason == PodStatusReason_ProviderFailed {
+			continue
+		}
+
+		status, err := s.provider.GetPodStatus(ctx, pod.Namespace, pod.Name)
+		if err != nil {
+			log.G(ctx).WithField("pod", pod.Name).Error("Error retrieving pod status")
```
Comment (robbiezhang) [inline vkubelet/pod.go]: WithField("namespace", pod.Namespace)

File: vkubelet/pod.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+	addPodAttributes(span, pod)
+
+	var delErr error
+	if delErr = s.provider.DeletePod(ctx, pod); delErr != nil && errors.IsNotFound(delErr) {
+		span.SetStatus(trace.Status{Code: trace.StatusCodeUnknown, Message: delErr.Error()})
+		return delErr
+	}
+	span.Annotate(nil, "Deleted pod from provider")
+
+	logger := log.G(ctx).WithField("pod", pod.Name)
```
Comment (robbiezhang) [inline vkubelet/pod.go]: WithField("namespace", pod.Namespace)

File: vkubelet/vkubelet.go [hunk 4]
```diff
@@ -194,34 +135,19 @@ func (s *Server) Run(ctx context.Context) error {
 			return pkgerrors.Wrap(err, "failed to watch pods")
 		}
 
-	loop:
 		for {
```
Comment (robbiezhang) [inline vkubelet/vkubelet.go]: feels like this for loop should happen in the watchForPodEvent function.

**Comment by cpuguy83** [comment]
Updated and all green.

**Pull request merged**